gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionContext.java

```diff
@@ -75,30 +82,51 @@
   private final Set<String> pendingMetricAggregationSourcesForOrderBy = new HashSet<>();
   private boolean sortAndPaginationNodeAdded = false;
 
-  // map of filter, selections (attribute, metrics, aggregations), order by attributes to source map
+  // map of filter, selections (attribute, metrics, aggregations), group bys, order by attributes to
+  // source map
   private final Map<String, Set<String>> allAttributesToSourcesMap = new HashMap<>();
 
-  private ExecutionContext(
+  public ExecutionContext(
```

Comment:
Not clean. Looks like we changed this for the purpose of reusing some part of the code

Reply:
Actually no. I just got rid of the other static `from` method required to instantiate execution context